index.bs

```diff
@@ -5565,20 +5551,83 @@ WebDriver [=extension command=] is defined as follows:
+            </tr>
+            <tr>
+                <td>|hasResidentKey|</td>
+                <td>boolean</td>
+                <td>[TRUE], [FALSE]</td>
+                <td>[FALSE]</td>
+            </tr>
+            <tr>
+                <td>|hasUserVerification|</td>
+                <td>boolean</td>
+                <td>[TRUE], [FALSE]</td>
+                <td>[FALSE]</td>
+            </tr>
+            <tr>
+                <td>|isUserPresent|</td>
+                <td>boolean</td>
+                <td>[TRUE], [FALSE]</td>
+                <td>[TRUE]</td>
+            </tr>
+            <tr>
+                <td>|isUserVerified|</td>
+                <td>boolean</td>
+                <td>[TRUE], [FALSE]</td>
+                <td>[TRUE]</td>
+            </tr>
+        </tbody>
+    </table>
+</figure>
+
 The [=remote end steps=] are:
 
  1. If |parameters| is not a JSON [=Object=], return a [=WebDriver error=] with [=WebDriver error code=]
      [=invalid argument=].
- 1. Let |protocol| be the result of [=trying=] to get the |parameter|'s {{AddVirtualAuthenticatorParameters/protocol}}
-    property.
- 1. If |protocol| is null, return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
- 1. Let |transport| be the result of [=trying=] to get the |parameter|'s {{AddVirtualAuthenticatorParameters/transport}}
-    property.
- 1. If |transport| is null, return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
- 1. Generate an arbitrary and unique string |authenticatorId|.
- 1. Create a new [=Virtual Authenticator=] with the fields |authenticatorId|, |protocol|, |transport|,
-    |hasResidentKey|, |hasUserVerification|, |isUserPresent|, and |isUserVerified|.
- 1. Store the authenticator in the [=Virtual Authenticator Database=].
+ 1. Let |authenticator| be a new [=Virtual Authenticator=].
+ 1. For each enumerable [=own property=] in |parameters|:
+    1. Let |key| be the name of the property.
+    1. Let |value| be the result of [=getting a property=] named |name| from |parameter|.
+    1. If there is no matching `key` for |key| in the [=Authenticator Configuration=], return a [=WebDriver error=] with
+        [=WebDriver error code=] [=invalid argument=].
+    1. If |value| is not one of the `valid values` for that |key|, return a [=WebDriver error=] with [=WebDriver error code=]
+        [=invalid argument=].
+    1. [=Set a property=] |key| to |value| on |authenticator|.
+ 1. For each property in [=Authenticator Configuration=] with a default defined:
+    1. If `key` is not a defined property of |authenticator|, [=set a property=] `key` to `default` on |authenticator|.
+ 1. For each property in [=Authenticator Configuration=]:
+    1. If `key` is not a defined property of |authenticator|, return a [=WebDriver error=] with [=WebDriver error code=]
+        [=invalid argument=].
+ 1. Generate an unique string |authenticatorId| out of Unreserved Characters [[RFC3986]].
```

Comment:
s/ out of Unreserved Characters  [[RFC3986]]. / using characters from the `unreserved` production defined in [Appendix A](https://tools.ietf.org/html/rfc3986#appendix-A) of [[RFC3986]]. /

Reply:
I missed this in the refactor - moved this text to the definition of authenticatorId.